### PR TITLE
Putting device path check in storageops for GCE

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**
- throwing error for DeviceMappings if unable to get actual device path
- throwing new error ErrInvalidDevicePath for device path check errors

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/storageops/gce/gce.go [hunk 1]
```diff
@@ -651,6 +677,46 @@ func (s *gceOps) getDisksFromAllZones(labels map[string]string) (map[string]*com
...
+		retryCount++
+		if retryCount >= devicePathMaxRetryCount {
+			break
+		}
+		time.Sleep(devicePathRetryInterval)
+	}
+	return "", err
+}
+
+func (s *gceOps) diskIDToBlockDevPath(devPath string) (string, error) {
```
Comment (adityadani) [inline pkg/storageops/gce/gce.go]: @harsh-px If this function is called only for storageops perspective, should it return an error if it is unable to find a symbolic link to an actual device path?

File: pkg/storageops/gce/gce.go [hunk 2]
```diff
@@ -597,7 +620,10 @@ func (s *gceOps) waitForAttach(
 	devicePath, err := task.DoRetryWithTimeout(
 		func() (interface{}, bool, error) {
 			devicePath, err := s.DevicePath(disk.Name)
-			if err != nil {
+			if se, ok := err.(*storageops.StorageError); ok &&
```
Comment (adityadani) [inline pkg/storageops/gce/gce.go]: In the DevicePath API(), we are not determining whether a volume is attached on remote node, but inferring it if it the disk shows as attached and the instance does not show any reference for this disk.

From GCE perspective is it safe to assume that, Disk's status changing and the Disk appearing in DesribeInstance output an atomic operation?

File: pkg/storageops/gce/gce.go [hunk 3]
```diff
@@ -259,7 +266,15 @@ func (s *gceOps) DeviceMappings() (map[string]string, error) {
 			continue
 		}
 
-		m[fmt.Sprintf("%s%s", googleDiskPrefix, d.DeviceName)] = path.Base(d.Source)
+		pathByID := fmt.Sprintf("%s%s", googleDiskPrefix, d.DeviceName)
+		devPath, err := s.diskIDToBlockDevPath(pathByID)
```
[the remark this answers is not in the document]
Comment (piyush-nimbalkar) [reply pkg/storageops/gce/gce.go]: DeviceMappings is called to get properties. 
- This is called after the drive is attached and verified.
- If this fails, we return the DefaultProvider properties anyway
Hence did not want to delay the initialization.

File: pkg/storageops/gce/gce.go (hunk 1 above)
Comment (harsh-px) [reply pkg/storageops/gce/gce.go]: Yes, this function is expected to pass for all paths (symbolic and regular). So any error would be unexpected and not ignored.

If storageops fails to find the dev path and does not return error, the path will not be useful to callers who expected a block device path.

File: pkg/storageops/gce/gce.go (hunk 2 above)
Comment (harsh-px) [reply pkg/storageops/gce/gce.go]: Discussed offline. This can be improved to just use disk users.

**Pull request merged**